stubs/redis/METADATA.toml

```
@@ -3,7 +3,15 @@ upstream_repository = "https://github.com/redis/redis-py"
 # Requires a version of cryptography with a `py.typed` file
 requires = ["cryptography>=35.0.0", "types-pyOpenSSL"]
 partial_stub = true
-obsolete_since = "5.0.0" # Released on 2023-08-15
+# While upstream added a py.typed file, the upstream annotations are rather
+# incomplete. By using "obsolete_since" we recommend deinstalling this
+# package, which isn't the best idea at the moment.
+# obsolete_since = "5.0.0" # Released on 2023-08-15
+extra_description = """\
+    Redis-py 5.0.0 added a py.typed file, but the annotations are fairly \
+    incomplete, which is why the typeshed maintainers recommend to continue \
+    using `types-redis` for the time being.\
```

It's hard to get the wording right here without making it sound like we're making a dig at the Redis maintainers, but maybe this is slightly less opinionated language:

```suggestion
    Note: Redis-py 5.0.0 added a py.typed file, but the inline annotations \
    are incomplete. Continuing to use `types-redis` for the time being may \
    lead to superior results.\
```